Sources/ProjectDescription/Dependency.swift

```diff
@@ -3,10 +3,10 @@ import Foundation
 /// Dependency contains the description of any kind of dependency of your Xcode project.
 public struct Dependency: Codable, Equatable {
     /// Name of the dependency
-    let name: String
+    public let name: String
 
     /// Type of requirement for the given dependency
-    let requirement: Dependency.Requirement
+    public let requirement: Dependency.Requirement
```

Comment:
Wonder if we could make this requirement specific to the Dependency manager. That way you wouldn't need to throw an error for carthage range versions. 

It feels odd that the API allows this.

Reply:
yea I have the same feeling

What do you think @pepibumur @facumenzella  ?

Reply:
Sounds good! I'm always in favor of using the type-system to validate the manifest. Linting should be the very last resource.